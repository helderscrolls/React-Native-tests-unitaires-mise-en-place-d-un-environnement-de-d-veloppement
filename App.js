import React from 'react';
import { StyleSheet, Text, View, Button, Alert } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default () => {
  const showAlert = () => {
    Alert.alert('Touch√©');
  };

  return (
    <View style={styles.container}>
      <Text>Articles</Text>
      <Button onPress={showAlert} title="Ajouter un article" />
    </View>
  );
};
