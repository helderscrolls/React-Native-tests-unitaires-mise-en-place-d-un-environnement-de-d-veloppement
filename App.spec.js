import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';

import App from './App';

describe('App component', () => {
  it('renders View', () => {
    const component = renderer.create(<App />);
    const instance = component.root;
    expect(instance.findByType(View)).toBeTruthy();
  });
});
