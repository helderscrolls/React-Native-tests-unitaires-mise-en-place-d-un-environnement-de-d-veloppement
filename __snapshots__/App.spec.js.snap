// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders View 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Articles
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4ba37b",
        "borderRadius": 50,
        "fontSize": 20,
        "fontWeight": "bold",
        "height": 30,
        "justifyContent": "center",
        "lineHeight": 30,
        "marginTop": 50,
        "opacity": 1,
        "width": 150,
      }
    }
  >
    <Text>
      Ajouter un article
    </Text>
  </View>
</View>
`;
